common4j/src/main/com/microsoft/identity/common/java/logging/DiagnosticContext.java

```diff
@@ -76,8 +78,8 @@ public IRequestContext getRequestContext() {
     public String getThreadCorrelationId() {
         IRequestContext context = getRequestContext();
         String correlationId = context.get(DiagnosticContext.CORRELATION_ID);
-        if (correlationId == null) {
-            correlationId = UNSET;
+        if (correlationId == null || correlationId.equals(UNSET)) {
+            correlationId = UUID.randomUUID().toString();
```

Comment:
Am I right in saying: 
it's okay to create a UUID here, because:
1) we don't need to use/set the thread ID, because that's not what's being used in the request headers (i.e. when setting the request headers, we use map key `DiagnosticContext.CORRELATION_ID`, not `DiagnosticContext.THREAD_ID`
2) `CommandDispatcher.initializeDiagnosticContext()` does the same; generate a UUID.

Reply:
Yes, that's correct.